File: Task_3/task_3.py
import math

# Функция для вычисления суммы и произведения цифр числа
def calculate_sum_and_product(n):
    sum_digits = 0
    product_digits = 1
    number_string = str(n) # Преобразуем число в строку

    for digit_char in number_string:
        digit = int(digit_char) # Преобразуем символ цифры в число
        sum_digits += digit
        product_digits *= digit
    return sum_digits, product_digits

if __name__ == "__main__":
    # Ввод чисел из стандартного ввода
    numbers = input("Введите числа через пробел: ").split()
    numbers = [int(n) for n in numbers if n.isdigit()] # Преобразуем строки в числа

    # Обрабатываем каждое введенное число
    for n in numbers:
        sum_digits, product_digits = calculate_sum_and_product(n) # Получаем сумму и произведение
        difference = abs(sum_digits - product_digits) # Вычисляем модуль разности

        # Формируем вывод разложения
        number_str = str(n)
        sum_expression = ' + '.join(number_str)
        product_expression = ' * '.join(number_str)
        print(f"{sum_digits} {product_digits} {difference} ({sum_expression} = {sum_digits}; {product_expression} = {product_digits}; |{sum_digits} - {product_digits}| = {difference})")

File: Task_3/task_3.ts
// Функция для вычисления суммы и произведения цифр числа
function calculateSumAndProduct(n: number): { sum: number; product: number } {
    let sum = 0;
    let product = 1;
    const numberString = n.toString(); // Преобразуем число в строку

    for (const digitChar of numberString) {
        const digit = parseInt(digitChar, 10); // Преобразуем символ цифры в число
        sum += digit;
        product *= digit;
    }
    return { sum, product };
}

// Функция для получения ввода от пользователя
function getNumbers(): number[] {
    const input = prompt('Введите числа через пробел:');
    if (!input) {
        return [];
    }
    return input.split(' ').map(Number).filter(n => !isNaN(n));
}

// Функция для вывода результатов
function printResults(numbers: number[]) {
    for (const n of numbers) {
        const { sum, product } = calculateSumAndProduct(n); // Получаем сумму и произведение
        const difference = Math.abs(sum - product); // Вычисляем модуль разности

        // Формируем вывод разложения
        const numberStr = n.toString();
        const sumExpression = numberStr.split('').map(Number).join(' + ');
        const productExpression = numberStr.split('').map(Number).join(' * ');

        console.log(`${sum} ${product} ${difference} (${sumExpression} = ${sum}; ${productExpression} = ${product}; |${sum} - ${product}| = ${difference})`);
    }
}

function main() {
    const numbers = getNumbers();
    printResults(numbers);
}

main();
